cmd/erasure-metadata-utils.go

```diff
@@ -153,28 +153,44 @@ func readVersionFromDisks(ctx context.Context, disks []StorageAPI, bucket, objec
 	return metadataArray, g.Wait()
 }
 
-// Return disks ordered by the meta.Erasure.Index information.
-func shuffleDisksByIndex(disks []StorageAPI, metaArr []FileInfo) (shuffledDisks []StorageAPI) {
+func shuffleDisksAndPartsMetadataByIndex(disks []StorageAPI, metaArr []FileInfo, distribution []int) (shuffledDisks []StorageAPI, shuffledPartsMetadata []FileInfo) {
 	shuffledDisks = make([]StorageAPI, len(disks))
+	shuffledPartsMetadata = make([]FileInfo, len(disks))
+	var inconsistent int
 	for i, meta := range metaArr {
 		if disks[i] == nil {
 			continue
 		}
+		// check if erasure distribution order matches the index
+		// position if this is not correct we discard the disk
+		// and move to collect others
+		if distribution[i] != meta.Erasure.Index {
+			inconsistent++ // keep track of inconsistent entries
+			continue
+		}
 		shuffledDisks[meta.Erasure.Index-1] = disks[i]
+		shuffledPartsMetadata[meta.Erasure.Index-1] = metaArr[i]
 	}
-	return shuffledDisks
+	if inconsistent == len(disks) {
```

Comment:
if one or more disks are down, this wont be equal

Reply:
@krishnasrinivas do you want me to capture that as well?